update installation instructions, add link to forum (#3890)
FuelLabs/sway: docs/book/src/introduction/installation.md

[`fuelup`](https://github.com/FuelLabs/fuelup) is the equivalent of Rust's `rustup` for the Fuel toolchain. It enables easily downloading binary releases of the Fuel toolchain.

1. Start by installing `fuelup` with the following command:

   ```sh
   curl --proto '=https' --tlsv1.2 -sSf \
   https://fuellabs.github.io/fuelup/fuelup-init.sh | sh
   ```

   This downloads the `fuelup-init` script to a temp directory on your machine, which installs `fuelup`. `fuelup-init` will ask for permission to add `~/.fuelup/bin` to your PATH. Otherwise, you can also pass `--no-modify-path` so that `fuelup-init` does not modify your PATH:

   ```sh
   curl --proto '=https' --tlsv1.2 -sSf \
   https://fuellabs.github.io/fuelup/fuelup-init.sh | sh -s -- --no-modify-path
   ```

2. Once fuelup is installed, fuelup-init automatically runs `fuelup toolchain install latest` to install the latest toolchain

>    You can run `fuelup toolchain install latest` at anytime to get the most up-to-date toolchain.

Reviewer: ```suggestion
   You can run `fuelup update` at anytime to get the most up-to-date toolchain.
```

I'd prefer to encourage `fuelup update` here, which updates all official toolchains that a user has installed, although `fuelup toolchain install latest` also works

   Unpacking and moving forc-wallet to /Users/user/.fuelup/toolchains/beta-2-aarch64-apple-darwin/bin
   Installed forc-wallet v0.1.2 for toolchain 'beta-2-aarch64-apple-darwin'

   Adding component fuel-core v0.15.3 to 'beta-2-aarch64-apple-darwin'
   Fetching binary from https://github.com/FuelLabs/fuel-core/releases/download/v0.15.3/fuel-core-0.15.3-aarch64-apple-darwin.tar.gz
   Unpacking and moving fuel-core to /Users/user/.fuelup/toolchains/    beta-2-aarch64-apple-darwin/bin
   Installed fuel-core v0.15.3 for toolchain 'beta-2-aarch64-apple-darwin'

   Adding component fuel-indexer v0.1.13 to 'beta-2-aarch64-apple-darwin'
   Fetching binary from https://github.com/FuelLabs/fuel-indexer/releases/download/v0.1.13/fuel-indexer-0.1.13-aarch64-apple-darwin.tar.gz
   Unpacking and moving fuel-indexer to /Users/user/.fuelup/toolchains/beta-2-aarch64-apple-darwin/bin
   Installed fuel-indexer v0.1.13 for toolchain 'beta-2-aarch64-apple-darwin'

   Installed:
   - forc 0.31.1
   - forc-explore 0.28.1
   - forc-wallet 0.1.2
   - fuel-core 0.15.3
   - fuel-indexer 0.1.13

   The Fuel toolchain is installed and up to date
   ```

You're all set to start building!

> #### Need Help?

Reviewer 2: I think this is the cause of the lint check failing, content looks good to me 👍 

```suggestion
### Need Help?
```

> Check out the tooling section in the [Fuel forum](https://forum.fuel.network/) if you're running into problems through the installation process. If you don't see your question, post the issue you're running into with as many details as possible and the team will get back to you asap!

Reviewer: There probably should also be a link to the fuelup book here: https://fuellabs.github.io/fuelup/latest/, so that people know where to look to self-troubleshoot issues before posting on the forum

```suggestion
You may refer to [The Fuelup Book](https://fuellabs.github.io/fuelup/latest/) for an in-depth look into fuelup, or check out the tooling section in the [Fuel forum](https://forum.fuel.network/) if you're running into problems through the installation process. If you don't see your question, post the issue you're running into with as many details as possible and the team will get back to you asap!
```
